From d77dacf686d07e2659ed1c7fc1f5dee85835926d Mon Sep 17 00:00:00 2001
From: bma <Bo.Ma@windriver.com>
Date: Fri, 31 May 2013 11:01:32 +0800
Subject: Fix assetion fail on class member check

Class member mInputFrameData of SoftMPEG4Encoder
could be used to store metadata, it might be allocated
before encoder initialized, reset it before using it
avoids asset fail.

For:AXIA-2586
Change-Id: Iad5d76e14f50ef32418f3abb5647aefec252499d
Signed-off-by: bma <Bo.Ma@windriver.com>
---
 .../codecs/m4v_h263/enc/SoftMPEG4Encoder.cpp       |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/codecs/m4v_h263/enc/SoftMPEG4Encoder.cpp b/media/libstagefright/codecs/m4v_h263/enc/SoftMPEG4Encoder.cpp
index 8bc0275..8f47605 100644
--- a/media/libstagefright/codecs/m4v_h263/enc/SoftMPEG4Encoder.cpp
+++ b/media/libstagefright/codecs/m4v_h263/enc/SoftMPEG4Encoder.cpp
@@ -149,7 +149,10 @@ OMX_ERRORTYPE SoftMPEG4Encoder::initEncParams() {
 
     if (mVideoColorFormat == OMX_COLOR_FormatYUV420SemiPlanar) {
         // Color conversion is needed.
-        CHECK(mInputFrameData == NULL);
+        /* 'mInputFrameData' could be used to store metadata,
+           it might be allocated before encoder initialized,
+           reset it before using it. */
+        delete mInputFrameData;
         mInputFrameData =
             (uint8_t *) malloc((mVideoWidth * mVideoHeight * 3 ) >> 1);
         CHECK(mInputFrameData != NULL);
-- 
1.7.4.1

